crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -554,6 +565,13 @@ fn optional_visibility() -> impl NoirParser<AbiVisibility> {
     })
 }
 
+fn optional_distinctness() -> impl NoirParser<AbiDistinctness> {
```

Comment:
There is a test function in this file named "parse_function" can you add a test case for distinct?